### PR TITLE
enable resp3 to leader: dont proxy new hello command

### PR DESCRIPTION
**Pull request by sjpotter**

### CODE REVIEW

**Comment by sjpotter** [comment]
ok, digging into redis code, there's RM_GetContextFlags() with a flag that is REDISMODULE_CTX_FLAGS_RESP3

**Comment by yossigo** [comment]
@sjpotter I think follower proxy is currently an incomplete feature, which mostly serves the ability to run Jepsen. I think the first goal is to have proper RESP3 support in normal mode.

**Comment by sjpotter** [comment]
I've done cursory tests (redis-cli -3) and I'm unsure what's not working once I added hello support.

**Comment by sjpotter** [comment]
this depends on https://github.com/redis/redis/pull/9634 being committed (or something like it)

File: raft.c [hunk 1]
```diff
@@ -1240,9 +1240,7 @@ RRStatus RedisRaftInit(RedisModuleCtx *ctx, RedisRaftCtx *rr, RedisRaftConfig *c
     }
 
     /* Cluster configuration */
-    if (rr->config->sharding) {
-        ShardingInfoInit(rr);
-    }
+    ShardingInfoInit(rr);
```
[the remark this answers is not in the document]
Comment (yossigo) [reply raft.c]:
```
127.0.0.1:6379> cluster slots
(error) ERR This instance has cluster support disabled
```

[the remark this answers is not in the document]
Comment (sjpotter) [reply raft.c]: changed to redis syntax

**Review by yossigo — APPROVED** [review]

**Pull request merged**